### PR TITLE
debug mode

### PR DESCRIPTION
**Pull request by passichenko**
Changes:
- I decided to wrap `slog` because it's the most promising and flexible way to do logging in go. For now we use global logger, so any otel bridge or custom handler will be applied to lekko logs. Later we can expose a way to set lekko-specific logger.
- Added `LogInfo` and `LogError` that wrap `slog.Info` and `slog.Error` respectfully and have the same API
- It has to be in a separate package `pkg/debug` so it can be used both inside the SDK and in generated code.

Logging events:
- log that debug mode is enabled
- On initialization, print repo, opts and error (in case of failure)
- evaluation
- fallback

Examples:
```
❯ : LEKKO_DEBUG=true go run main.go
2024/07/08 15:06:43 INFO LEKKO_DEBUG=true, running in debug mode
2024/07/08 15:06:43 INFO LEKKO_API_KEY environment variable is not set, in-code fallback will be used
2024/07/08 15:06:49 INFO Lekko fallback result=ROOT
```

```
2024/07/08 15:25:03 INFO LEKKO_DEBUG=true, running in debug mode
2024/07/08 15:25:03 INFO Connected to Lekko repository=passichenko-lekko-1/lekko-configs opts=[{APIKey:lekko_3ff48a26-7d69-4868-b019-a46df6fafb87_5663e76f-040e-48c6-bf52-497a4bb824a3}]
2024/07/08 15:25:04 INFO Lekko evaluation name=default/title context=map[env:production] result="value:\"Hello from Lekko!\""
2024/07/08 15:25:04 INFO Lekko evaluation name=default/root-message context=map[] result="value:\"MY ROOT\""
```

### CODE REVIEW

**Comment by passichenko** [comment]
* **#104** <a href="https://app.graphite.dev/github/pr/lekkodev/go-sdk/104?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @passichenko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by konradjniemiec** [comment]
let's ship for now, but I'm worried that we are requiring go 1.21, 1.20 is still getting security updates

File: pkg/debug/debug.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package debug
+
+import (
+	"log/slog"
+	"os"
+)
+
+func LogInfo(msg string, args ...any) {
+	if isDebugMode {
+		slog.Info(msg, args...)
```
Comment (DavidSGK) [inline pkg/debug/debug.go:10]: Since it's debug mode, should we log to debug instead? :)

**Review by DavidSGK — APPROVED** [review]
Also worried about the Go versions, but down to ship - we should maybe update our docs to explicitly call out the version

**Comment by passichenko** [comment]
> let's ship for now, but I'm worried that we are requiring go 1.21, 1.20 is still getting security updates

Good point! I think we can follow officially supported versions:
> Each major Go release is supported until there are two newer major releases.
from https://go.dev/doc/devel/release

So, based on that 1.20 is not officially supported (last minor was in Feb 2024). And 1.24 is coming out soon:
> Go 1.23 is expected to be released in August 2024

File: pkg/debug/debug.go (hunk 1 above)
Comment (passichenko) [reply pkg/debug/debug.go:10]: Yeah, good point. I was even thinking to change it to "log level", but let's for now make "debug_mode = true" => "log_level = debug".

Comment (passichenko) [reply pkg/debug/debug.go:30]: I just realized that there is a small problem with this. The default level is `info` and we probably don't want to change it for the global logger.

**Pull request merged**